File: Serenity.Script.UI/FilterPanel/FilterDialog.cs
using jQueryApi;
using jQueryApi.UI.Widgets;
using System;
using System.Collections.Generic;
using System.Html;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true)]
    public class FilterDialog : TemplatedDialog
    {
        private FilterPanel filterPanel;

        public FilterDialog()
        {
            filterPanel = new FilterPanel(this.ById("FilterPanel"));
            filterPanel.ShowInitialLine = true;
            filterPanel.ShowSearchButton = false;
            filterPanel.UpdateStoreOnReset = false;
        }

        protected override string GetTemplate()
        {
            return "<div id='~_FilterPanel'/>";
        }

        protected override DialogOptions GetDialogOptions()
        {
            var opt = base.GetDialogOptions();
            opt.Buttons = new[] 
            {
                new DialogButton {
                    Text = Q.Text("Dialogs.OkButton"),
                    Click = delegate
                    {
                        filterPanel.Search();
                        if (filterPanel.HasErrors)
                        {
                            Q.NotifyError(Q.Text("Controls.FilterPanel.FixErrorsMessage"));
                            return;
                        }
                        DialogClose();
                    }
                },
                new DialogButton {
                    Text = Q.Text("Dialogs.CancelButton"),
                    Click = DialogClose
                }
            };
            opt.Title = Q.Text("Controls.FilterPanel.DialogTitle");

            return opt;
        }

        public FilterPanel FilterPanel
        {
            get { return filterPanel; }
        }
    }
}

File: Serenity.Scripts/CoreLib/UI/Filtering/FilterDialog.ts
namespace Serenity {

    @Decorators.registerClass('Serenity.FilterDialog')
    export class FilterDialog extends TemplatedDialog<any> {

        private filterPanel: FilterPanel;

        constructor() {
            super();

            this.filterPanel = new FilterPanel(this.byId('FilterPanel'));
            this.filterPanel.set_showInitialLine(true);
            this.filterPanel.set_showSearchButton(false);
            this.filterPanel.set_updateStoreOnReset(false);
        }

        get_filterPanel(): FilterPanel {
            return this.filterPanel;
        }

        protected getTemplate(): string {
            return '<div id="~_FilterPanel"/>';
        }

        protected getDialogOptions() {
            var opt = super.getDialogOptions();

            opt.buttons = [
                {
                    text: Q.text('Dialogs.OkButton'),
                    click: () => {
                        this.filterPanel.search();
                        if (this.filterPanel.get_hasErrors()) {
                            Q.notifyError(Q.text('Controls.FilterPanel.FixErrorsMessage'), '', null);
                            return;
                        }

                        this.dialogClose();
                    }
                },
                {
                    text: Q.text('Dialogs.CancelButton'),
                    click: () => this.dialogClose()
                }
            ];

            opt.title = Q.text('Controls.FilterPanel.DialogTitle');

            return opt;
        }
    }

}